proposals/nnnn-rpc-design-refactoring.md

```diff
@@ -0,0 +1,153 @@
+# RPC design refactoring
+
+* Proposal: [SDL-NNNN](nnnn-rpc-design-refactoring.md)
+* Author: [Alexander Kutsan](https://github.com/LuxoftAKutsan)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core]
+
+## Introduction
+
+This proposal is to rework design of application manager layer in order to:
+ - Reduce coupling between components
+ - Reduce responsibility of ApplicationManager class
+ - Provide plugin architecture for easy adding additional functionality and RPC groups   
```

Comment:
Provide plugin architecture for easy adding additional functionality and RPC groups 
should become
Provide a plugin architecture for adding additional functionality and RPC groups easily.

Reply:
@mrapitis, done in 2c9f7a3f7bb7e6c8b26bb1e4aec779cf65733666

Reply:
@AStasiuk looks like this one still needs to be updated.  It may have been changed while you were working on it..